src/services/download/mod.rs

```diff
@@ -191,22 +193,25 @@ impl DownloadService {
 
 /// Download the source from a stream.
 ///
-/// This is common functionality used by all many downloaders.
+/// This is common functionality used by many downloaders.
 async fn download_stream(
     source: impl Into<ObjectFileSource>,
-    stream: impl Stream<Item = Result<impl AsRef<[u8]>, DownloadError>>,
+    mut stream: impl Stream<Item = Result<impl AsRef<[u8]>, DownloadError>> + Unpin,
```

Comment:
Can we just leave the pin mut internally? It seems like an implementation detail of the function.